### PR TITLE
Signature verifier

### PR DESCRIPTION
**Pull request by kubaplas**
Needs to be rebased after other prs are merged for tests to pass

### CODE REVIEW

File: core/src/contract_env.rs [hunk 1]
```diff
@@ -101,4 +104,15 @@ impl ContractEnv {
...
     }
+
+    pub fn verify_signature(
+        &self,
+        message: &Bytes,
+        signature: &Bytes,
+        public_key: &PublicKey
+    ) -> bool {
+        let (signature, _) = casper_types::crypto::Signature::from_bytes(signature.as_slice())
+            .unwrap_or_else(|_| self.revert(ExecutionError(CouldntDeserializeSignature)));
```
Comment (kpob) [inline core/src/contract_env.rs]:
```suggestion
            .unwrap_or_else(|_| self.revert(CouldntDeserializeSignature));
```
revert() accepts Into<OdraError> so we don't need to wrap the error

File: core/src/crypto.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    for i in 0..amount {
+        // Create keypair.
+        let secret_key = SecretKey::ed25519_from_bytes([i; 32]).unwrap();
+        let public_key = PublicKey::from(&secret_key);
+
+        // Create an AccountHash from a public key.
+        let account_addr = AccountHash::from(&public_key);
+
+        // Create a GenesisAccount.
+        accounts.insert(account_addr.try_into().unwrap(), (secret_key, public_key));
```
Comment (kpob) [inline core/src/crypto.rs]: As @zie1ony stated https://github.com/odradev/odra/pull/272#discussion_r1413704292 core functions should not fail (except reverts)

Comment (kubaplas) [reply core/src/crypto.rs]: Well, if account hash cannot be converted into address, something is terribly wrong. Nevertheless, I pushed a version which continues in this case.

**Review by kpob — APPROVED** [review]

File: core/src/crypto.rs (hunk 2 above)
Comment (kpob) [reply core/src/crypto.rs]: I'm not sure if skipping is the best strategy, we just swallow an error. Maybe we should panic or revert depending on the context. A few lines above `let secret_key = SecretKey::ed25519_from_bytes([i; 32]).unwrap()` you unwrap a SecretKey creation. We could handle it better, but agree, very likely it'll never fail. I'm not insisting, just sayin'.

Comment (kubaplas) [reply core/src/crypto.rs]: Now it panics :)

Comment (kpob) [reply core/src/crypto.rs]: If used in tests only then fine. :)

**Pull request merged**